src/VisualStudio/Core/Def/Implementation/ProjectSystem/MiscellaneousFilesWorkspace.cs

```diff
@@ -328,15 +329,40 @@ private void AttachToDocument(uint docCookie, string moniker)
             // This should always succeed since we only got here if we already confirmed the moniker is acceptable
             var languageInformation = TryGetLanguageInformation(moniker);
             Contract.ThrowIfNull(languageInformation);
-            var parseOptions = languageInformation.ParseOptions;
 
-            if (Path.GetExtension(moniker) == languageInformation.ScriptExtension)
+            var languageServices = Services.GetLanguageServices(languageInformation.LanguageName);
+            var compilationFactory = languageServices.GetService<ICompilationFactoryService>();
+            var compilationOptions = compilationFactory.GetDefaultCompilationOptions();
+            var syntaxTreeFactory = languageServices.GetService<ISyntaxTreeFactoryService>();
+            var parseOptions = syntaxTreeFactory.GetDefaultParseOptions();
```

Comment:
It doesn't look like we use the MiscFilesWorkspace in VS. However we don't implement the `ICompilationFactoryService`, or the `ISyntaxTreeFactoryService`, so I suggest bailing from this method early in case either is null or at least put in a bunch of asserts so we can discover what's happening when there is a crash.

FYI @vladima and @minestarks